inst/include/interface/rcpp/rcpp_objects/rcpp_fleet.hpp

```diff
@@ -81,7 +88,156 @@ class FleetInterface : public FleetInterfaceBase {
+            for (size_t i = 0; i < this->derived_index.size(); i++) {
+                this->derived_index[i] = fleet->expected_index[i];
+            }
+
+        }
+
+    }
+
+    virtual std::string to_json() {
+        std::stringstream ss;
+
+        ss << "\"module\" : {\n";
+        ss << " \"name\" : \"Fleet\",\n";
+
+        ss << " \"type\" : \"fleet\",\n";
+        ss << " \"tag\" : \"" << this->name << "\",\n";
+        ss << " \"id\": " << this->id << ",\n";
+
+        ss << " \"parameter\": {\n";
+        ss << " \"name\": \"log_Fmort\",\n";
+        ss << " \"id\":" << this->log_Fmort.id_m << ",\n";
+        ss << " \"type\": \"vector\",\n";
+        ss << " \"values\": " << this->log_Fmort << "\n},\n";
+
+        ss << " \"parameter\": {\n";
+        ss << " \"name\": \"log_Fmort\",\n";
+        ss << " \"id\":" << this->log_q.id_m << ",\n";
+        ss << " \"type\": \"vector\",\n";
+        ss << " \"values\": " << this->log_q << "\n},\n";
+
+
+        ss << " \"derived_quantity\": {\n";
+        ss << "  \"name\": \"cnaa\",\n";
+        ss << "  \"values\":[";
+        if (this->derived_cnaa.size() == 0) {
+            ss << "]\n";
+        } else {
+            for (size_t i = 0; i < this->derived_cnaa.size() - 1; i++) {
+                ss << this->derived_cnaa[i] << ", ";
+            }
+            ss << this->derived_cnaa[this->derived_cnaa.size() - 1] << "]\n";
+        }
+        ss << " },\n";
+
+        ss << " \"derived_quantity\": {\n";
+        ss << "  \"name\": \"cwaa\",\n";
+        ss << "  \"values\":[";
+        if (this->derived_cwaa.size() == 0) {
+            ss << "]\n";
+        } else {
+            for (size_t i = 0; i < this->derived_cwaa.size() - 1; i++) {
+                ss << this->derived_cwaa[i] << ", ";
+            }
+            ss << this->derived_cwaa[this->derived_cwaa.size() - 1] << "]\n";
+        }
+        ss << " },\n";
+
+
+        ss << " \"derived_quantity\": {\n";
+        ss << "  \"name\": \"age_composition \",\n";
+        ss << "  \"values\":[";
+        if (this->derived_age_composition.size() == 0) {
+            ss << "]\n";
+        } else {
+            for (size_t i = 0; i < this->derived_age_composition.size() - 1; i++) {
+                ss << this->derived_age_composition[i] << ", ";
+            }
+            ss << this->derived_age_composition[this->derived_age_composition.size() - 1] << "]\n";
+        }
+        ss << " },\n";
+
+        ss << " \"derived_quantity\": {\n";
+        ss << "  \"name\": \"index \",\n";
+        ss << "  \"values\":[";
+        if (this->derived_index.size() == 0) {
+            ss << "]\n";
+        } else {
+            for (size_t i = 0; i < this->derived_index.size() - 1; i++) {
+                ss << this->derived_index[i] << ", ";
+            }
+            ss << this->derived_index[this->derived_index.size() - 1] << "]\n";
+        }
+        ss << " },\n";
+
+        return ss.str();
```

Comment:
do we need to return the ss here - is this printing to the R console?

Reply:
There were cout statements left that was causing the printing this has been fixed.